sqlx-mysql/src/any.rs

```diff
@@ -7,7 +7,7 @@ use either::Either;
 use futures_core::future::BoxFuture;
 use futures_core::stream::BoxStream;
 use futures_util::{stream, StreamExt, TryFutureExt, TryStreamExt};
-use sqlx_core::any::{
+pub use sqlx_core::any::{
```

Comment:
`pub use` here causes IDEs to suggest the wrong imports: https://github.com/launchbadge/sqlx/pull/3254#issuecomment-2144043823

The fix there was to mark `sqlx_postgres::any` as `#[doc(hidden)]` but in hindsight I think we should have just re-scoped the exports.

`pub(crate) use sqlx_core::any::*;` should be fine. 

Alternatively, just import directly from `sqlx_core::any` in the other files. The blanket re-export was only a monkey patch to get the code compiling after [extracting the driver crates from sqlx-core](https://github.com/launchbadge/sqlx/pull/2039).

Reply:
[rewrite use and address implementation concerns](https://github.com/launchbadge/sqlx/pull/3608/commits/b5c279abedf0a82741b7f1802e17479ca2a64f3b)